Hibernate Reactive with Oracle after bump to 2.2.1 throws casting exception - cannot cast DeleteOrUpsertOperation to OptionalTableUpdate

### Describe the bug

https://github.com/quarkusio/quarkus/pull/38162 is causing issue in my Hibernate Reactive app using Oracle database. I inspected:

- https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.7#jpa-hibernate-orm
- https://github.com/hibernate/hibernate-orm/blob/6.3/migration-guide.adoc
- https://github.com/hibernate/hibernate-orm/blob/6.4.0/migration-guide.adoc
- https://hibernate.org/reactive/releases/2.2/

without success.

### Expected behavior

Works.

### Actual behavior

Exception is thrown and app startup fails:

```
20:15:35,049 INFO  [app] 20:15:33,870 Failed to start application (with profile [prod]): java.lang.RuntimeException: Failed to start quarkus
20:15:35,049 INFO  [app] 	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
20:15:35,049 INFO  [app] 	at io.quarkus.runtime.Application.start(Application.java:101)
20:15:35,049 INFO  [app] 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
20:15:35,050 INFO  [app] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
20:15:35,050 INFO  [app] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
20:15:35,050 INFO  [app] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
20:15:35,050 INFO  [app] 	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
20:15:35,050 INFO  [app] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
20:15:35,050 INFO  [app] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
20:15:35,051 INFO  [app] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
20:15:35,051 INFO  [app] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
20:15:35,051 INFO  [app] Caused by: jakarta.persistence.PersistenceException: Unable to build EntityManagerFactory
20:15:35,051 INFO  [app] 	at io.quarkus.hibernate.reactive.runtime.FastBootHibernateReactivePersistenceProvider.createEntityManagerFactory(FastBootHibernateReactivePersistenceProvider.java:102)
20:15:35,051 INFO  [app] 	at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
20:15:35,051 INFO  [app] 	at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
20:15:35,052 INFO  [app] 	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:157)
20:15:35,052 INFO  [app] 	at io.quarkus.hibernate.orm.runtime.JPAConfig$1.run(JPAConfig.java:64)
20:15:35,052 INFO  [app] 	at java.base/java.lang.Thread.run(Thread.java:1583)
20:15:35,052 INFO  [app] Caused by: java.lang.ClassCastException: class org.hibernate.sql.model.jdbc.DeleteOrUpsertOperation cannot be cast to class org.hibernate.sql.model.internal.OptionalTableUpdate (org.hibernate.sql.model.jdbc.DeleteOrUpsertOperation and org.hibernate.sql.model.internal.OptionalTableUpdate are in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @46f7f36a)
20:15:35,052 INFO  [app] 	at org.hibernate.reactive.persister.entity.impl.ReactiveMergeCoordinatorStandardScopeFactory.createOperation(ReactiveMergeCoordinatorStandardScopeFactory.java:107)
20:15:35,052 INFO  [app] 	at org.hibernate.reactive.persister.entity.impl.ReactiveMergeCoordinatorStandardScopeFactory.createOperationGroup(ReactiveMergeCoordinatorStandardScopeFactory.java:56)
20:15:35,052 INFO  [app] 	at org.hibernate.persister.entity.mutation.UpdateCoordinatorStandard.buildStaticUpdateGroup(UpdateCoordinatorStandard.java:1646)
20:15:35,053 INFO  [app] 	at org.hibernate.persister.entity.mutation.UpdateCoordinatorStandard.<init>(UpdateCoordinatorStandard.java:90)
20:15:35,053 INFO  [app] 	at org.hibernate.persister.entity.mutation.MergeCoordinator.<init>(MergeCoordinator.java:23)
20:15:35,053 INFO  [app] 	at org.hibernate.reactive.persister.entity.impl.ReactiveMergeCoordinatorStandardScopeFactory.<init>(ReactiveMergeCoordinatorStandardScopeFactory.java:34)
20:15:35,053 INFO  [app] 	at org.hibernate.reactive.persister.entity.impl.ReactiveCoordinatorFactory.buildMergeCoordinator(ReactiveCoordinatorFactory.java:56)
20:15:35,053 INFO  [app] 	at org.hibernate.reactive.persister.entity.impl.ReactiveSingleTableEntityPersister.buildMergeCoordinator(ReactiveSingleTableEntityPersister.java:107)
20:15:35,053 INFO  [app] 	at org.hibernate.persister.entity.AbstractEntityPersister.doLateInit(AbstractEntityPersister.java:3415)
20:15:35,054 INFO  [app] 	at org.hibernate.persister.entity.AbstractEntityPersister.postInstantiate(AbstractEntityPersister.java:3679)
20:15:35,054 INFO  [app] 	at org.hibernate.metamodel.model.domain.internal.MappingMetamodelImpl.finishInitialization(MappingMetamodelImpl.java:201)
20:15:35,054 INFO  [app] 	at org.hibernate.internal.SessionFactoryImpl.initializeMappingModel(SessionFactoryImpl.java:364)
20:15:35,054 INFO  [app] 	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:293)
20:15:35,054 INFO  [app] 	at org.hibernate.reactive.session.impl.ReactiveSessionFactoryImpl.<init>(ReactiveSessionFactoryImpl.java:28)
20:15:35,054 INFO  [app] 	at io.quarkus.hibernate.reactive.runtime.boot.FastBootReactiveEntityManagerFactoryBuilder.build(FastBootReactiveEntityManagerFactoryBuilder.java:31)
20:15:35,055 INFO  [app] 	at io.quarkus.hibernate.reactive.runtime.FastBootHibernateReactivePersistenceProvider.createEntityManagerFactory(FastBootHibernateReactivePersistenceProvider.java:92)
20:15:35,055 INFO  [app] 	... 5 more
```

### How to Reproduce?

Reproducer:

1. `git@github.com:michalvavrik/quarkus-test-suite.git`
2. `quarkus-test-suite`
3. `mvn clean verify -f sql-db/hibernate-reactive/ -Dit.test=OracleDatabaseIT`

### Output of `uname -a` or `ver`

Fedora 38

### Output of `java -version`

21

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.4

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -103,7 +103,7 @@
         <hibernate-orm.version>6.4.2.Final</hibernate-orm.version>
         <bytebuddy.version>1.14.7</bytebuddy.version> <!-- Version controlled by Hibernate ORM's needs -->
         <hibernate-commons-annotations.version>6.0.6.Final</hibernate-commons-annotations.version> <!-- version controlled by Hibernate ORM -->
-        <hibernate-reactive.version>2.2.1.Final</hibernate-reactive.version>
+        <hibernate-reactive.version>2.2.2.Final</hibernate-reactive.version>
         <hibernate-validator.version>8.0.1.Final</hibernate-validator.version>
         <!-- When updating, align hibernate-search.version-for-documentation in docs/pom.xml -->
         <hibernate-search.version>7.0.0.Final</hibernate-search.version>

